### PR TITLE
Fix balance_battle crash path on invalid team_sizes

### LINKED ISSUE #430: [Bug]: (Enum.EmptyError) empty error during balance_battle request
### Describe the Bug

Logs:
```
Request: POST /teiserver/api/spads/balance_battle
** (exit) an exception was raised:
    ** (Enum.EmptyError) empty error
        (elixir 1.14.5) lib/enum.ex:1855: anonymous fn/0 in Enum.max/1
        (teiserver 0.1.0) lib/teiserver_web/controllers/api/spads_controller.ex:164: TeiserverWeb.API.SpadsController.balance_battle/2
        (teiserver 0.1.0) lib/teiserver_web/controllers/api/spads_controller.ex:1: TeiserverWeb.API.SpadsController.action/2
        (teiserver 0.1.0) lib/teiserver_web/controllers/api/spads_controller.ex:1: TeiserverWeb.API.SpadsController.phoenix_controller_pipeline/2
        (phoenix 1.7.6) lib/phoenix/router.ex:430: Phoenix.Router.__call__/5
        (teiserver 0.1.0) lib/teiserver_web/endpoint.ex:1: TeiserverWeb.Endpoint.plug_builder_call/2
        (teiserver 0.1.0) lib/teiserver_web/endpoint.ex:1: TeiserverWeb.Endpoint.call/2
        (plug_cowboy 2.6.1) lib/plug/cowboy/handler.ex:11: Plug.Cowboy.Handler.init/2
```

```ex
          team_size =
            balance_result.team_sizes
            |> Map.values()
            |> Enum.max()
            ```
It seems `team_sizes` might be empty.

### Reproduce the bug

Unfortunately the logs don't contain any other details. Don't know the team count, team size or balance algorithm used. There are only 2 such errors recently in logs so I think it's probably some edge case.

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by veez-e**
## Summary
- harden SpadsController.balance_battle/2 to derive team dimensions through a safe helper
- avoid crashing on malformed balance results by returning empty.json when team sizes are invalid
- make is_non_empty_balance_result?/1 robust against nil/missing/wrongly-typed 	eam_sizes
- add regression tests for empty/malformed 	eam_sizes and valid-dimensions extraction

## Testing
- MIX_ENV=test mix test test/teiserver_web/controllers/api/spads_controller_test.exs

Fixes #430

### CODE REVIEW

File: lib/teiserver_web/controllers/api/spads_controller.ex [hunk 1]
```diff
@@ -209,11 +201,20 @@ defmodule TeiserverWeb.API.SpadsController do
...
+         true <- is_integer(team_size) and team_size > 0 do
+      {:ok, {map_size(team_sizes), team_size}}
+    else
+      _ -> :error
     end
   end
+
+  def is_non_empty_balance_result?(balance_result) do
+    match?({:ok, _}, get_balance_team_dimensions(balance_result))
+  end
```
Comment (L-e-x-o-n) [inline lib/teiserver_web/controllers/api/spads_controller.ex]: This seems unused now.

**Comment by L-e-x-o-n** [comment]
Thanks
